dolphin/drivers/dellemc_storage/vmax/alert_handler.py

```diff
@@ -0,0 +1,159 @@
+               Probable values: Critical, Major, Minor, Warning, Info
+    device_alert_sn : Sequence number of alert generated. This will be helpful
+                      during alert clearing process
+    manufacturer : Vendor of the device
+    Product_name : Name of the product
+    probable_cause : Probable reason for alert generation
+    clear_type : Alarm clearance type such as manual, automatic, reset clear etc
+    me_category : Resource category of the device generating the alarm
+                  Probable value: Network,Server,Storage..        
+    """
+
+    def parse_alert(self, context, alert):
+        """Parse alert data got from alert manager and fill the alert model."""
+
+        # Check for mandatory alert attributes
+        for attr in self.mandatory_alert_attributes:
+            if not alert.get(attr):
+                msg = "Mandatory information %s missing in alert message. " \
+                      % attr
+                raise exception.ValidationError(detail=msg)
+
+        alert_model = {}
+        # These information are sourced from device registration info
+        alert_model['me_dn'] = alert['storage_id']
+        alert_model['me_name'] = alert['storage_name']
+        alert_model['manufacturer'] = alert['vendor']
+        alert_model['product_name'] = alert['model']
+
+        # trap info do not contain occur time, update with received time
+        # Get date and time. Format will be like : Wed May 20 01:53:29 2020
+        curr_time = datetime.now()
+        alert_model['occur_time'] = curr_time.strftime('%c')
+
+        # Fill all the alert model fields
+        if alert.get('category'):
+            alert_model['category'] = alert['category']
+        else:
+            alert_model['category'] = self.default_category
+
+        # Array id is used to fill unique id at source system side
+        alert_model['native_me_dn'] = alert['connUnitName']
+
+        # Location is name-value pair having component type and component name
+        if alert.get('emcAsyncEventComponentType') \
+                and alert.get('emcAsyncEventComponentName'):
+            component_type = alert_mapper.component_type_mapping.get(
+                alert.get('emcAsyncEventComponentType'), "")
+            alert_model['location'] = 'Component type: ' \
+                                      + component_type \
+                                      + ',Component name: ' \
+                                      + alert['emcAsyncEventComponentName']
+        else:
+            alert_model['location'] = self.default_location
+
+        if alert.get('connUnitEventType'):
+            alert_model['event_type'] = alert['connUnitEventType']
+        else:
+            alert_model['event_type'] = self.default_event_type
+
+        if alert.get('connUnitEventSeverity'):
+            alert_model['severity'] = alert['connUnitEventSeverity']
+        else:
+            alert_model['severity'] = self.default_severity
+
```

Comment:
from the code logic, why the data of alert doesn't provide the eventtype and severity sometimes.

Reply:
This is to make sure that we always have these values filled